src/figuresClasses.ts

```diff
@@ -1,19 +1,104 @@
+export type Shape = 'triangle' | 'circle' | 'rectangle';
+export type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
+
+  getArea(): number;
+}
 
+export function roundArea(digit: number): number {
+  return Math.floor(digit * 100) / 100;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape;
+
+  color: Color;
+
+  a: number;
+
+  b: number;
 
+  c: number;
+
+  constructor(color: Color, a: number, b: number, c: number) {
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+
+    if (a <= 0
+      || b <= 0
+      || c <= 0
+      || a >= (b + c)
+      || b >= (a + c)
+      || c >= (a + b)) {
+      throw new Error('Resulting value should not be 0');
```

Comment:
I think it`ll be better to divide these cases into diff blocks and throw 2 diff errors

Reply:
Agree

Reply:
but i like this view;))))